pkg/csi/cinder/controllerserver.go

```diff
@@ -263,6 +261,9 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 	if err != nil {
 		klog.V(3).Infof("Failed to query for existing Snapshot during CreateSnapshot: %v", err)
 	}
+	// Snapshot Create
+	req.Parameters["cinder.csi.openstack.org/cluster"] = cs.Driver.cluster
```

Comment:
why you are passing it in request parameters?? parameters are defined in manifests and passed by kubernetes to csi driver. `cs.Driver.cluster` we are passing it as a flag

Reply:
please refer to line 94 above
properties := map[string]string{"cinder.csi.openstack.org/cluster": cs.Driver.cluster}

I am doing basically simliar way to volume create...
@adisky

Reply:
ya thats what i am saying, why assigning it to req.Parameters?

Reply:
@jichenjc I agree with what @adisky is asking here. In the function you reference parameters is a new temporary variable but the request reference in this method is being mutated with the parameters. Is there a reason for this?

Reply:
ok, I updated...